Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -141,120 +159,362 @@ final class ExecuteCommandTests: XCTestCase {
+
+          let url = try XCTUnwrap(
+            showDocumentRequestURI.value?.fileURL,
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          let fileContents = try String(contentsOf: url, encoding: .utf8)
+
+          XCTAssertEqual(
+            fileContents,
+            "// MyMacroClient.swift @ 5:3 - 5:20\n(1 + 2, \"1 + 2\")",
+            "File doesn't contain macro expansion. Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
+
+          XCTAssertEqual(
+            url.lastPathComponent,
+            "MyMacroClient.swift",
+            "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+          )
         }
+      }
+    }
+  }
 
-        @main
-        struct MyMacroPlugin: CompilerPlugin {
-          let providingMacros: [Macro.Type] = [
-            StringifyMacro.self,
-          ]
+  func testAttachedMacroExpansion() async throws {
+    try await SkipUnless.canBuildMacroUsingSwiftSyntaxFromSourceKitLSPBuild()
+
+    let files: [RelativeFileLocation: String] = [
+      "MyMacros/MyMacros.swift": #"""
+      import SwiftCompilerPlugin
+      import SwiftSyntax
+      import SwiftSyntaxBuilder
+      import SwiftSyntaxMacros
+
+      public struct DictionaryStorageMacro {}
+
+      extension DictionaryStorageMacro: MemberMacro {
+        public static func expansion(
+          of node: AttributeSyntax,
+          providingMembersOf declaration: some DeclGroupSyntax,
+          in context: some MacroExpansionContext
+        ) throws -> [DeclSyntax] {
+          return ["\n  var _storage: [String: Any] = [:]"]
         }
-        """#,
-        "MyMacroClient/MyMacroClient.swift": """
-        @freestanding(expression)
-        public macro stringify<T>(_ value: T) -> (T, String) = #externalMacro(module: "MyMacros", type: "StringifyMacro")
+      }
 
-        func test() {
-          1️⃣#2️⃣stringify3️⃣(1 + 2)
+      extension DictionaryStorageMacro: MemberAttributeMacro {
+        public static func expansion(
+          of node: AttributeSyntax,
+          attachedTo declaration: some DeclGroupSyntax,
+          providingAttributesFor member: some DeclSyntaxProtocol,
+          in context: some MacroExpansionContext
+        ) throws -> [AttributeSyntax] {
+          return [
+            AttributeSyntax(
+              leadingTrivia: [.newlines(1), .spaces(2)],
+              attributeName: IdentifierTypeSyntax(
+                name: .identifier("DictionaryStorageProperty")
+              )
+            )
+          ]
         }
-        """,
-      ],
-      manifest: SwiftPMTestProject.macroPackageManifest,
-      options: options
-    )
-    try await SwiftPMTestProject.build(at: project.scratchDirectory)
+      }
 
-    let (uri, positions) = try project.openDocument("MyMacroClient.swift")
+      public struct DictionaryStoragePropertyMacro: AccessorMacro {
```

Comment:
Picking up https://github.com/swiftlang/sourcekit-lsp/pull/1479#discussion_r1649078893 again:

Ah, I see. You need to enable background indexing (passing `enableBackgroundIndexing: true` to SwiftPMTestProject`) and remove the `try await SwiftPMTestProject.build(at: project.scratchDirectory)` call. Then it should work.

Reply:
I will try this.